core/src/main/java/org/hisp/dhis/android/core/program/internal/ProgramModuleDownloader.kt

```diff
@@ -61,53 +61,48 @@ internal class ProgramModuleDownloader @Inject constructor(
     private val optionCall: OptionCall,
     private val optionGroupCall: OptionGroupCall,
     private val programOrganisationUnitLinkStore: OrganisationUnitProgramLinkStore,
-) : UntypedModuleDownloader {
+) : UntypedModuleDownloaderCoroutines {
 
-    override fun downloadMetadata(): Completable {
-        return Completable.defer {
-            val orgUnitProgramUids = programOrganisationUnitLinkStore
-                .selectDistinctSlaves(OrganisationUnitProgramLinkTableInfo.Columns.PROGRAM)
+    override suspend fun downloadMetadata() {
+
+        val orgUnitProgramUids = programOrganisationUnitLinkStore
+            .selectDistinctSlaves(OrganisationUnitProgramLinkTableInfo.Columns.PROGRAM)
+
+        val programs = programCall.download(orgUnitProgramUids)
+
+        val programUids = getUids(programs)
+        val programStages = programStageCall.download(programUids)
+        val trackedEntityUids = ProgramParentUidsHelper.getAssignedTrackedEntityUids(programs)
+        val trackedEntityTypes = trackedEntityTypeCall.download(trackedEntityUids)
+
+        val attributes = trackedEntityAttributeCall.download(
+            ProgramParentUidsHelper
+                .getAssignedTrackedEntityAttributeUids(programs, trackedEntityTypes),
+        )
+
+
+        val optionSetUids = ProgramParentUidsHelper.getAssignedOptionSetUids(
+            attributes,
+            programStages,
+        )
+        flowOf(
```

Comment:
I think this `flowOf` doesn't add any value, does it? I know it follows the same structure than before, but I don't see the point in keeping this calls together. They don't run in parallel in any case, I we don't want them to run in parallel because of the database lock